sdk/core/Azure.Core/src/Serialization/IModelSerializable.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.Serialization
+{
+    /// <summary>
+    /// Indicates that the impleenter can be serialized and deserialized.
+    /// The format of the serialization is determined by the implementer.
+    /// </summary>
+    /// <typeparam name="T">The type to deserialize the value into.</typeparam>
+    public interface IModelSerializable<out T>
+    {
+        /// <summary>
+        /// Serializes the model into a <see cref="BinaryData"/>.
+        /// </summary>
+        /// <param name="options">The <see cref="ModelSerializerOptions"/> to use.</param>
+        /// <returns>A binary representation of the serialized model.</returns>
+        /// <exception cref="InvalidOperationException">If the model does not support the requested <see cref="ModelSerializerOptions.Format"/>.</exception>
```

Comment:
This will correspond to whatever is decided about the type of the format exception?

Reply:
good call out this should be updated to NotSupportedException as of right now, but yes it should change if we change that.

Reply:
https://github.com/Azure/azure-sdk-for-net/pull/38120